gigascience/gigadb-website
# A11y - Skip to content link

# Pull request for issue: #1362

This is a pull request for the following functionalities:

* Add a skip to main content link to all layouts. The purpose of skip links is to allow to skip header navigation quickly in every page and go directly to content, when navigating with the keyboard

## How to test?

- Verify that the first focusable element in each page is the skip to main link, you might check one page for each of the existing layouts, e.g.:
  - Home page http://gigadb.gigasciencejournal.com:9170 (new_main layout)
  - Result detail http://gigadb.gigasciencejournal.com:9170/dataset/100006 (new_dataset layout)
  - FAQ http://gigadb.gigasciencejournal.com:9170/site/faq (new_faq layout)
  - Admin home http://gigadb.gigasciencejournal.com:9170/site/admin (main layout)
  - Admin inner page http://gigadb.gigasciencejournal.com:9170/adminDatasetAuthor/admin (datasetpage layout)
- Verify that the skip to main link is only displayed when it receives focus

## How have functionalities been implemented?

A link is added to every layout. This link links to the main element. The link is displayed only on focus, otherwise, it's hidden from view

## Any issues with implementation?

None

## Any changes to automated tests?

None

## Any changes to documentation?

None

## Any technical debt repayment?

None

## Any improvements to CI/CD pipeline?

None
